src/platform/file_manager.c

```diff
@@ -83,10 +85,159 @@ typedef const char * dir_name;
 #include <unistd.h>
 #endif
 
+#ifdef __vita__
```

Comment:
Can we move the caching parts to a separate file - `file_manager_cache.c` perhaps? It's quite a lot of code, only necessary for vita, and it's separate functionality ([single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)).

Reply:
I was thinking whether I should do that actually. Guess I'll do it then :wink: